Funcky/Monads/Lazy/LazyExtensions.cs

```diff
@@ -0,0 +1,10 @@
+using System.Diagnostics.CodeAnalysis;
+using static System.Diagnostics.CodeAnalysis.DynamicallyAccessedMemberTypes;
+
+namespace Funcky.Monads;
+
+public static partial class LazyExtensions
+{
+    public static Lazy<T> Flatten<[DynamicallyAccessedMembers(PublicParameterlessConstructor)] T>(this Lazy<Lazy<T>> lazy)
+        => new Lazy<T>(() => lazy.Value.Value);
```

Comment:
Nitpick: Could use `Lazy.FromFunc`:
```suggestion
        => Lazy.FromFunc(() => lazy.Value.Value);
```